parsigs/parse_sig_api.py

```diff
@@ -207,19 +209,44 @@ def _create_structured_sig(model_entities, drug=None, form=None):
             structured_sig.form = _to_singular(text)
         elif label == 'Frequency':
             structured_sig.frequencyType = _get_frequency_type(text)
-            structured_sig.interval = _get_interval(text)
+            interval_text = _get_string_after_keyword(text, "every")
+            if len(interval_text) > 0:
+                structured_sig.interval = _get_amount_from_frequency_tags(interval_text)
+            else:
+                structured_sig.interval = 1
+
+            times_text = _get_string_until_keyword(text, "times")
+            structured_sig.times = _get_amount_from_frequency_tags(times_text)
```

Comment:
Lets see if I understand correctly.

Given a sig like: "Take 3 tablets 2 times a day"

First, the NER model extracts the "Frequency" part as: "2 times a day".
Then, the `times_text` would be `"2"`, and `_get_amount_from_frequency_tags` would return the first digit found, which is `2`.

Reply:
Exactly :)